src/evm/useEvmTokenBalances.ts

```diff
@@ -28,11 +28,12 @@ export const useEvmTokenBalances = (
 
   const memoizedParams = useDeepMemo(() => params, params);
   const apiKey = useGetApiKey();
-  const baseUrl = useGetBaseUrl();
+  const proxyUrl = useGetProxyUrl();
 
   // Function to fetch data for a specific page
   const fetchDataAsync = async (offset: string | null) => {
-    if (!apiKey || !walletAddress || !isAddress(walletAddress)) return;
+    if ((!apiKey && !proxyUrl) || !walletAddress || !isAddress(walletAddress))
+      return;
```

Comment:
we should prob throw legit errors here. It took me a second to realize what was going on. Or at least console.error